delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -114,8 +108,20 @@ private String validateAndRetrieveContent(@Nullable HttpURLConnection urlConn) t
                                      getError(urlConn));
     }
 
-    try (Reader reader = new InputStreamReader(urlConn.getInputStream(), StandardCharsets.UTF_8)) {
-      return CharStreams.toString(reader);
+    try (InputStream inputStream = urlConn.getInputStream()) {
+      String contentType = urlConn.getContentType();
```

Comment:
it's not good to rely on the content type to determine the return type. Should always use whatever type is passed in.

For bytes, there should be a separate method with a fixed byte[] return type. You can still share most of the code by doing something like:

```

private <T> T retrieve(HttpURLConnection urlConn, Function<HttpURLConnection, T> decodeFunc) {

}

private <T> T retrieve(HttpURLConnection urlConn, Type type) {
  return retrieve(urlConn, conn -> { // use gson to decode })
}

private byte[] retrieve(HttpURLConnection urlConn) {
  return retrieve(urlConn, conn -> { // use BytesStreams to decode })
}
```